text/0478-tracked-properties-updates.md

```diff
@@ -0,0 +1,1131 @@
+    set(this, 'friends', friends);
+  }
+
+  @computed('friends.[]')
+  get friendNames() {
+    return this.friends.map(friend => friend.name);
+  }
+}
+
+let joey = new Person(
+  emberA([
+    { name: 'Phoebe' },
+    { name: 'Monica' },
+    { name: 'Chandler' },
+    { name: 'Ross' },
+  ])
+);
+
+// Using pushObject will cause `friendNames` to update
+joey.friends.pushObject({ name: 'Rachel' });
+
+// Alternatively, we can update the whole array:
+set(joey, 'friends', [...joey.friends, { name: 'Rachel' }]);
+```
+
+If the property is tracked, then `set` is not necessary, and the field can be
+updated directly as you would with normal tracked properties:
+
+```js
+import { computed } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+class Person {
+  @tracking friends;
+
+  constructor(friends = []) {
+    this.friends = friends;
+  }
+
+  @computed('friends.[]')
+  get friendNames() {
+    return this.friends.map(friend => friend.name);
+  }
+}
+
+let joey = new Person([
+  { name: 'Phoebe' },
+  { name: 'Monica' },
+  { name: 'Chandler' },
+  { name: 'Ross' },
+]);
+
+joey.friends = [...joey.friends, { name: 'Rachel' }];
+```
+
+#### Computed Properties and Tracking
+
+Computed properties will autotrack when they are accessed from templates or
+through other getters, like tracked properties.:
+
+```js
+import { computed, watchable } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+class Person {
+  @tracked firstName;
+  @tracked lastName;
+
+  constructor(firstName, lastName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+  }
+
+  @computed('firstName', 'lastName')
+  get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+
+  set fullName(value) {
+    let [firstName, lastName] = value.split(' ');
+
+    this.firstName = firstName;
+    this.lastName = lastName;
+  }
+
+  // legalName will update whenever `fullName` updates
+  get legalName() {
+    return this.fullName;
+  }
+}
+```
+
+### When to Use `get` and `set`
+
+Ember's classic change tracking system used two methods to ensure that all data
+was accessed properly and updated correctly: `get` and `set`.
+
+```js
+import { get, set } from '@ember/object';
+
+let person = {};
+
+set(person, 'firstName', 'Amy');
+set(person, 'lastName', 'Lam');
+
+get(person, 'firstName'); // 'Amy'
+get(person, 'lastName'); // 'Lam'
+```
+
+In classic Ember, all property access had to go through these two methods. Over
+time, these rules have become less strict, and now they have been minimized to
+just a few cases. In general, in a modern Ember app, you shouldn't need to use
+them all that much. As long as you are marking your properties as `@tracked`,
+autotracking should automatically figure out what needs to change, and when.
+
+However, there still are two cases where you _will_ need to use them:
+
+- When accessing and updating plain, undecorated properties on objects
+- When using Ember's `ObjectProxy` class
```

Comment:
Missing a third case: manual proxies that implement `unknownProperty` and `setUnknownProperty`